File: java/src/main/java/com/alxarafe/app/domain/topology/entity/Warehouse.java
package com.alxarafe.app.domain.topology.entity;

import com.alxarafe.app.domain.topology.valueobject.WarehouseCode;
import com.alxarafe.app.domain.topology.valueobject.WarehouseId;

import java.util.Objects;

/**
 * Aggregate Root representing a physical warehouse.
 */
public final class Warehouse {

    private final WarehouseId id;
    private final WarehouseCode code;
    private final String name;

    public Warehouse(WarehouseId id, WarehouseCode code, String name) {
        this.id = Objects.requireNonNull(id, "Warehouse id cannot be null.");
        this.code = Objects.requireNonNull(code, "Warehouse code cannot be null.");
        Objects.requireNonNull(name, "Warehouse name cannot be null.");
        if (name.isBlank()) {
            throw new IllegalArgumentException("Warehouse name cannot be empty.");
        }
        this.name = name;
    }

    public WarehouseId id() {
        return id;
    }

    public WarehouseCode code() {
        return code;
    }

    public String name() {
        return name;
    }
}

File: php/src/Domain/Topology/Entity/Warehouse.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Domain\Topology\Entity;

use Alxarafe\App\Domain\Topology\ValueObject\WarehouseCode;
use Alxarafe\App\Domain\Topology\ValueObject\WarehouseId;
use InvalidArgumentException;

/**
 * Aggregate Root representing a physical warehouse.
 */
final readonly class Warehouse
{
    public function __construct(
        private WarehouseId $id,
        private WarehouseCode $code,
        private string $name,
    ) {
        if ($name === '') {
            throw new InvalidArgumentException('Warehouse name cannot be empty.');
        }
    }

    public function id(): WarehouseId
    {
        return $this->id;
    }

    public function code(): WarehouseCode
    {
        return $this->code;
    }

    public function name(): string
    {
        return $this->name;
    }
}
